pkg/broker/ingress/handler.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/cloudevents/sdk-go/v2/protocol/pubsub"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+// DecoupleSink is an interface to send events to a decoupling sink (e.g., pubsub).
+type DecoupleSink interface {
+	// Send sends the event to a decoupling sink.
+	Send(ctx context.Context, event event.Event) error
+}
+
+// handler receives events and persists them to storage (pubsub).
+type handler struct {
+	// inbound is the cloudevents client to receive events.
+	inbound client.Client
+	// decouple is the client to send events to a decouple sink.
+	decouple DecoupleSink
+}
+
+func NewHandler(ctx context.Context, options ...Option) (*handler, error) {
+	h := &handler{}
+	var err error
+	// Create a default HTTP client for ingress.
+	if h.inbound, err = client.NewDefault(); err != nil {
```

Comment:
Default clients should only be created if a client was not specified in options.

Reply:
Good call. Thanks.